notebooks/cytotable_and_pycytominer_analysis.py

```diff
@@ -0,0 +1,253 @@
+                - datetime.strptime(memray_data["metadata"]["start_time"], tformat)
+            ).total_seconds(),
+            "total_memory (bytes)": memray_data["total_bytes_allocated"],
+        }
+    )
+
+    # cleanup
+    pathlib.Path(target_bin).unlink(missing_ok=True)
+    pathlib.Path(target_json).unlink(missing_ok=True)
+
+df_results = pd.DataFrame(results)
+df_results
+# -
+
+df_results["data_input_renamed"] = df_results["data_input"].str.replace(
+    "all_cellprofiler", "input"
+)
+df_results["cytotable_time_duration (secs)"] = df_results[
+    df_results["file_input"] == "cytotable_convert_nf1.py"
+]["time_duration (secs)"]
+df_results["cytotable_total_memory (bytes)"] = df_results[
+    df_results["file_input"] == "cytotable_convert_nf1.py"
+]["total_memory (bytes)"]
+df_results["pycytominer_time_duration (secs)"] = df_results[
+    df_results["file_input"] == "pycytominer_merge_nf1.py"
+]["time_duration (secs)"]
+df_results["pycytominer_total_memory (bytes)"] = df_results[
+    df_results["file_input"] == "pycytominer_merge_nf1.py"
+]["total_memory (bytes)"]
+df_results = (
+    df_results.apply(lambda x: pd.Series(x.dropna().values))
+    .drop(["file_input", "time_duration (secs)", "total_memory (bytes)"], axis=1)
+    .dropna()
+)
+df_results
+
+# +
+# read time chart
+fig = px.line(
+    df_results,
+    y=[
+        "cytotable_time_duration (secs)",
+        "pycytominer_time_duration (secs)",
+    ],
+    x="data_input_renamed",
+    title="CytoTable and Pycytominer<br>SQLite Processing Time Comparison",
+    labels={"data_input_renamed": "Input File", "value": "Seconds"},
+    height=400,
+    symbol_sequence=["diamond"],
+    color_discrete_sequence=[
+        px.colors.qualitative.Vivid[6],
+        px.colors.qualitative.Vivid[4],
+    ],
+)
+
+# rename the lines for the legend
+newnames = {
+    "cytotable_time_duration (secs)": "CytoTable",
+    "pycytominer_time_duration (secs)": "Pycytominer",
+}
+# referenced from: https://stackoverflow.com/a/64378982
+fig.for_each_trace(
+    lambda t: t.update(
+        name=newnames[t.name],
+        legendgroup=newnames[t.name],
+        hovertemplate=t.hovertemplate.replace(t.name, newnames[t.name]),
+    )
+)
+
+# update the legend
+fig.update_layout(
+    legend_title_text="",
+    legend=dict(x=0.01, y=0.98, bgcolor="rgba(255,255,255,0.8)"),
+    font=dict(
+        size=16,  # global font size
+    ),
+)
+# fig.update_xaxes(range=[-0.03, 5.2])
+fig.update_traces(mode="lines+markers")
+
+fig.write_image(join_read_time_image)
+fig.write_image(join_read_time_image.replace(".png", ".svg"))
+Image(url=join_read_time_image.replace(".png", ".svg"))
+
+# +
+# memory size
+
+fig = px.line(
+    df_results,
+    y=[
+        "cytotable_total_memory (bytes)",
+        "pycytominer_total_memory (bytes)",
+    ],
+    x="data_input_renamed",
+    title="CytoTable and Pycytominer<br>SQLite Total Memory Consumption",
+    labels={"data_input_renamed": "Input File", "value": "bytes"},
```

Comment:
It might be helpful to specify the size of the input file in the x axis (e.g., Input File (2GB)) this way people can understand more about what 16x actually means in practical terms

Furthermore, it might be helpful to specify the axis as `bytes (billions)` in addition to adding the letter after the number. Better yet would be to convert everything to GB IMHO

Reply:
Thanks, these are great points, I'll incorporate these in.